File: Códigos/COJ/Estructura de Datos/2637-Range_Sum_Query.cpp
/**
 * Problem: Range Sum Query
 * Problem ID: 2637
 * url: http://coj.uci.cu/24h/problem.xhtml?pid=2637
 */

#include <stdio.h>
#include <stdlib.h>

#define MAX 100002

int main(int argc, char const *argv[]) {
	long long int list[MAX];
	int ncases, n, q, i, j, num;
	list[0] = 0;

	scanf("%d", &ncases);

	while(ncases--) {
		scanf("%d %d", &n, &q);
		for(int k = 1; k <=n; k++) {
			scanf("%d", &num);
			list[k] = list[k-1] + num;
		}

		while(q--) {
			scanf("%d %d", &i, &j);
			if (i>j) {
				puts("0");
			} else {
				printf("%lld\n", list[j+1] - list[i]);
			}
		}

		if (ncases) puts("");
	}
	return 0;
}

File: src/COJ/Estructura de Datos/2637-Range_Sum_Query.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-

#
# Problem: Range Sum Query
# Problem ID: 2637
# url: http://coj.uci.cu/24h/problem.xhtml?pid=2637
#

import sys


def getLine():
    return sys.stdin.readline().strip()

ncases = int(getLine())

while ncases:
    ncases -= 1

    getLine()
    nelements, nqueries = map(int, getLine().split())

    ls = map(int, getLine().split())
    for i in range(1, nelements):
        ls[i] += ls[i-1]

    for k in range(nqueries):
        i, j = map(int, getLine().split())
        print ls[j] if i == 0 else ls[j] - ls[i-1]

    if ncases:
        print
